local_pathfinding/node_navigate.py

```diff
@@ -77,14 +78,17 @@ def __init__(self):
             timer_period_sec=pub_period_sec, callback=self.desired_heading_callback
         )
 
+        planner = self.get_parameter("path_planner").get_parameter_value().string_value
+        self.get_logger().debug(f"Got parameter: {planner=}")
+
         # attributes from subscribers
         self.ais_ships = None
         self.gps = None
         self.global_path = None
         self.filtered_wind_sensor = None
 
         # attributes
-        self.local_path = LocalPath(parent_logger=self.get_logger())
+        self.local_path = LocalPath(parent_logger=self.get_logger(), planner=planner)
```

Comment:
Rather than making planner an init argument make it an argument of local_path.update_if_needed. And to match with the rest of the code make planner an instance variable of `LocalPathState` and `OMPLPathState`